### PR TITLE
[Spec] Infrastructure Spec

### PR DESCRIPTION
**Pull request by golddove**
Contains the original spec for the infrastructure effort, with the following changes/additions:

* Additional steps added to ReleaseChecklist.md
* Crossed out completed parts of the Infrastructure spec
* Added Pipelines.md, which lists all pipelines in active use

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6232)

### CODE REVIEW

File: specs/devops/Infrastructure.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+* Typically, a stable release needs to be made for every rendering library and the website on a monthly cadence. Some rendering libraries have multiple component packages, so each changed package needs to be released.
+* Ahead of release, each package needs to be independently versioned, so manual PRs are created to modify version numbers defined in config files (i.e. `gradle.properties`, `custom.props`, and `package.json`).
+* Smoke testing is manually performed on the latest available apps created by the nightly pipelines, checking for any major visible regressions in `samples/Scenario` cards. When an app is signed off by a tester, the corresponding rendering library is approved for release.
+* A release pipeline is defined for 4 platforms and the website.
+    * Android pipeline is fully automated to publish
+    * iOS pipeline is fully automated to publish
+        * iOS packages are not signed
+    * .NET piplines are fully automated to publish
+    * Web rendering libraryis fully automated to publish
```
Comment (almedina-ms) [inline specs/devops/Infrastructure.md:32]: typo: library is

File: specs/devops/Infrastructure.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		* JS packages are not signed
+    * Website pipeline is fully automated to release
+* Release notes are manually compiled
+    * List of issues that were fixed since last stable release is compiled for each platform. Ideally, this involves only going through issues tagged with the current milestone. In reality, some issues are fixed without changing milestones, while some fixes are issue-less PRs, which must all be manually checked
+    * Notes must be written with issue URLs linked, to trigger bot comment on issues (notifies users who are waiting for fix to be available)
+    * Non-blocking issues that arose during smoke are manually added to release notes' known issues
```
Comment (almedina-ms) [inline specs/devops/Infrastructure.md:38]: during smoke tests

File: specs/devops/Infrastructure.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+Additionally, the repository contains libraries contributed and maintained by the community, which we do not currently validate or publish in any way.
+
+## Motivation
+
+As long as releases have significant overhead, the planning process becomes encumbered by the release process. There are several points throughout the current process that have room for improvement. Primarily, monthly releases involve significant context-switching by all developers, with many manual steps that take time away from development and open the door for errors.
+* Some platforms do not publish consumable apps nightly, requiring a local build to smoke test before release.
+* Versioning must be done manually and differently for each platform, which results in several last-minute PRs that must all go through 
+* Dependencies are audited and fixed during release, which introduces changes after smoke tests and increases risk
+* Release note generated is a highly manual process that not only takes time, but frequently misses some fixes. This has resulted in hosts being unsure of what changes they are consuming in an upgrade, and voicing lack of confidence in a release.
```
Comment (almedina-ms) [inline specs/devops/Infrastructure.md:48]: release notes generation may be a better noun

**Review by paulcam206 — APPROVED** [review]
:shipit:

**Review by almedina-ms — APPROVED** [review]

**Pull request merged**